netx/conn.go

```diff
@@ -0,0 +1,150 @@
+package netx
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/sirupsen/logrus"
+)
+
+// InstanceProvider is the channel where updated list of desired service are published
+type InstanceProvider <-chan []net.Addr
+
+// Sender sends payload to given address
+type Sender func(addr net.Addr, payload []byte) (int, error)
```

Comment:
Why not use simple `io.Writer` here?

Reply:
We can't use `io.Writer` because an address is dynamically changed. That's why I introduced a new type.